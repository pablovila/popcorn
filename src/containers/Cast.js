import React from "react";
import PropTypes from "prop-types";

import CastCard from "components/Cards/CastCard";

const cast = ({ cast }) => {
  return (
    <div className="columns is-multiline">
      {cast.slice(0, 12).map(person => (
        <div className="column is-2" key={person.id}>
          <CastCard person={person} />
        </div>
      ))}
    </div>
  );
};

cast.propTypes = {
  cast: PropTypes.array.isRequired
};

export default cast;
