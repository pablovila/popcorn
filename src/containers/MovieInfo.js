import React from "react";
import PropTypes from "prop-types";

import { IMAGE_BASE_URL } from "apis/tmdb/apiConfig";
import Hero from "components/UI/Hero";
import Cast from "./Cast";

const movieInfo = ({ movie }) => {
  const date = movie.release_date ? movie.release_date.split("-") : null;
  const releaseYear = date ? date[0] : null;
  return (
    <>
      <Hero>
        <div className="media">
          <figure className="media-left">
            <p className="image">
              <img src={`${IMAGE_BASE_URL}/w300/${movie.poster_path}`} />
            </p>
          </figure>
          <div className="media-content">
            <div className="content">
              <h2 className="title is-2">
                {movie.title}{" "}
                <span className="title is-5 has-text-grey">
                  ({releaseYear})
                </span>
              </h2>
              <p>{movie.overview}</p>
              {movie.genres && movie.genres.length && (
                <div className="tags">
                  {movie.genres.map(genre => (
                    <span className="tag is-danger" key={genre.id}>
                      {genre.name}
                    </span>
                  ))}
                </div>
              )}
            </div>
          </div>
        </div>
      </Hero>
      {movie.cast && movie.cast.length && (
        <section className="section">
          <div className="container">
            <h4 className="title is-4">Reparto</h4>
            <Cast cast={movie.cast} />
          </div>
        </section>
      )}
    </>
  );
};

movieInfo.propTypes = {
  movie: PropTypes.object.isRequired
};

export default movieInfo;
