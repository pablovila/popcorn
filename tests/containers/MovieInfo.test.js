import React from "react";
import { shallow, mount } from "enzyme";

import MovieInfo from "containers/MovieInfo";
import Cast from "containers/Cast";

import { goodMovie, badMovie } from "./mocks/MovieInfo.mock";

describe("test MovieInfo container", () => {
  it("renders MovieInfo container", () => {
    const movieInfo = shallow(<MovieInfo movie={goodMovie} />);

    expect(movieInfo.find("h2").text()).toEqual(`${goodMovie.title} (2019)`);
    expect(movieInfo).toMatchSnapshot();
  });

  it("renders MovieInfo children and UI subcomponents", () => {
    const movieInfo = mount(<MovieInfo movie={goodMovie} />);

    expect(movieInfo.find(Cast)).toHaveLength(1);
    expect(movieInfo).toMatchSnapshot();
  });

  it("renders MovieInfo children even with a bad movie", () => {
    const movieInfo = mount(<MovieInfo movie={badMovie} />);

    expect(movieInfo.find(Cast)).toHaveLength(0);
    expect(movieInfo).toMatchSnapshot();
  });
});
