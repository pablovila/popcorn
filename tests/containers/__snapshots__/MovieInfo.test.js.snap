// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MovieInfo container renders MovieInfo children and UI subcomponents 1`] = `ReactWrapper {}`;

exports[`test MovieInfo container renders MovieInfo children even with a bad movie 1`] = `ReactWrapper {}`;

exports[`test MovieInfo container renders MovieInfo container 1`] = `ShallowWrapper {}`;
